### PR TITLE
Adding OrderReceiptPage

### LINKED ISSUE #424: UI: Receipts
### Designs

https://projects.invisionapp.com/share/KH11D3ZN53GC#/screens/462885076
https://projects.invisionapp.com/share/KH11D3ZN53GC#/screens/462972248

### Requirements

- [ ] The receipt page is a readonly version of the checkout page, we should strive to reuse code/components here as mouch as possible.

### Out of Scope
- [ ] The "Back to Order History" link can be skipped for now since we haven't done that page yet.

### PR DESCRIPTION
**Pull request by annagav**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots

- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
Fix #424

#### What's this PR do?
Add an order receipts page

#### How should this be manually tested?
Make sure to set up a purchasable product, create a fullfilled oder for it, got to the order summary page `/orders/history/` and click to view one of the orders.

Screen Shot 2022-03-07 at 9 31 44 AM
Screen Shot 2022-03-07 at 9 31 10 AM

### CODE REVIEW

**Comment by rhysyngsun** [comment]
>     2. Do we want to be able to reach the receipt page for a given order directly? This may be a better question for @rhysyngsun (and may be outside the scope of this request).

Yes, the receipt page should have a navigable url.

File: static/js/containers/pages/checkout/OrderReceiptPage.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      <CartItemCard
+        key={`itemcard_${orderItem.product.id}`}
+        product={orderItem.product}
+      />
+    )
+  }
+
+  renderOrderSummaryCard() {
+    const { orderReceipt, discounts } = this.props
+    const totalPaid = parseFloat(orderReceipt.total_price_paid)
```
Comment (jkachel) [inline static/js/containers/pages/checkout/OrderReceiptPage.js:45]: This needs a check for null - it fails for me consistently.

File: static/js/components/OrderSummaryCard.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                <h5>Total</h5>
+              </div>
+              <div className="ml-auto">
+                <h5>{fmtDiscountPrice || fmtPrice}</h5>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        {!SETTINGS.features.disable_discount_ui && !orderFulfilled && discountCode ? (
```
Comment (jkachel) [inline static/js/components/OrderSummaryCard.js:122]: The discountCode check here prevents ApplyCouponForm from rendering.

**Comment by annagav** [comment]
@jkachel I addressed all the comments. Thank you

File: static/js/containers/pages/checkout/OrderReceiptPage.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        </Loader>
+      </DocumentTitle>
+    )
+  }
+}
+
+const mapStateToProps = state => ({
+  currentUser:  state.entities.currentUser,
+  orderReceipt: state.entities.orderReceipt,
+  isLoading:    pathOr(true, ["queries", "orderReceipt", "isPending"], state)
```
Comment (jkachel) [inline static/js/containers/pages/checkout/OrderReceiptPage.js:135]: "orderReceipt" here should be the receiptQueryKey

File: static/js/components/OrderSummaryCard.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+export class OrderSummaryCard extends React.Component<Props> {
+  renderAppliedCoupons() {
+    const { discounts, clearDiscount, orderFulfilled } = this.props
+
+    if (discounts === null || discounts.length === 0) {
+      return null
+    }
+
+    let discountAmountText = null
+    const discountAmount = Number(discounts[0].amount)
```
Comment (jkachel) [inline static/js/components/OrderSummaryCard.js:30]: discounts[x] has a "redeemed_discount" key - as is this renders as NaN (and also has problems on line 32 and 58)

**Review by jkachel — CHANGES_REQUESTED** [review]
Couple more small things

**Comment by annagav** [comment]
@jkachel I addressed your comments. I think it is ready for review again

**Review by jkachel — APPROVED** [review]
LGTM 👍

**Pull request merged**